app/config/config.yml

```diff
@@ -16,6 +16,13 @@ parameters:
     mail_themes_uri: "/mails/themes"
     mail_themes_dir: "%kernel.project_dir%%mail_themes_uri%"
 
+# Autowires Core controllers
+services:
+    PrestaShopBundle\Controller\:
+        resource: '../../src/PrestaShopBundle/Controller/*'
+        exclude: '../../src/PrestaShopBundle/Controller/Api'
```

Comment:
This folder contains already-registered controllers as services so I exclude it (see my PR description)

Reply:
You can add this as a comment in the yaml file

Reply:
Can you use `%kernel.root_dir%` instead of relative paths?

Reply:
We could also declare the previous services as alias?

Reply:
> We could also declare the previous services as alias?

It does not seem possible because **these controllers use [property injection](https://symfony.com/doc/current/service_container/injection_types.html#property-injection)** to be constructed.
See for example the [API StockController](https://github.com/PrestaShop/PrestaShop/blob/develop/src/PrestaShopBundle/Resources/config/services/bundle/controller.yml#L12):
```
class StockController extends ApiController
{
    /**
     * @var StockRepository
     */
    public $stockRepository;

    /**
     * @var QueryStockParamsCollection
     */
    public $queryParams;

    /**
     * @var MovementsCollection;
     */
    public $movements;
```

I could replace this with a constructor and allow auto-wiring to do his job (if all necessary objects are correctly type-hinted) but this would be a BC break 🤔

Reply:
These services could be declared by class name as service name, then aliased to the former names, so that there is coherence in the naming scheme for all controllers.

In all truth, we should have declared service names instead of autowiring them, because I don't think we use that anywhere else.

Reply:
> These services could be declared by class name as service name, then aliased to the former names, so that there is coherence in the naming scheme for all controllers.

I see what you mean. Modify YAML file to use the classnames are service IDs, then add aliases. I can do that.

> In all truth, we should have declared service names instead of autowiring them, because I don't think we use that anywhere else.

You want me to close this PR and create one with a big YAML file which declares all controllers as services instead ? I can do it but then we'll have to be careful to always think about adding new controllers into it. The autowiring will always be up-to-date whereas the manual declaration will require human review to be always accurate.

Reply:
I know that autowiring is better, using class names is the de-facto standard in symfony because of autowiring.

> Modify YAML file to use the classnames are service IDs, then add aliases.

That's probably the best solution. Let's keep autowiring and classname services for controllers, and add deprecated aliases for API controllers.